src/main/scala/io/radicalbit/nsdb/connector/kafka/sink/NSDbSinkWriter.scala

```diff
@@ -162,32 +162,46 @@ object NSDbSinkWriter {
 
   private case class WriteFailureException(msg: String) extends RuntimeException(msg)
 
+  /**
+    * Manipulate future result according to semantic delivery policy
+    * @param semanticDelivery [[SemanticDelivery]] if not defined, at_most_once delivery is chosen
+    * @param fResult future result of nsdb grpc write
+    * @param maxRetries maximum number of retries if at_least_once semantic delivery is chosen
+    * @param retryInterval time interval between two retries if at least once semantic delivery is chosen
+    * @param timeout time to wait for future completing if at least once semantic delivery is chosen
+    */
   def writeWithDeliveryPolicy(semanticDelivery: Option[SemanticDelivery],
                               fResult: => Future[List[RPCInsertResult]],
                               maxRetries: Int,
                               retryInterval: Duration,
                               timeout: Duration) = {
 
-    val policy = RetryPolicies.limitRetries(maxRetries - 1)
+    val policy = RetryPolicies.limitRetries[Id](maxRetries - 1)
 
     val finiteDurationSleep: FiniteDuration = FiniteDuration(retryInterval._1, retryInterval._2)
 
-    def onFailure(throwable: Throwable, details: RetryDetails): Future[Unit] = {
-      logger.debug(throwable.getMessage)
-      Sleep[Future].sleep(finiteDurationSleep)
+    def wasSuccessful: Try[List[RPCInsertResult]] => Boolean =
+      t => t.map(_.forall(_.completedSuccessfully)).getOrElse(false)
+
+    def onFailure(fail: Try[List[RPCInsertResult]], details: RetryDetails): Unit = {
+      logger.debug("Retrying...")
```

Comment:
we can raise the log level to warn, what's your though ?

Reply:
strongly agree